### PR TITLE
Enable ConfigMap for MariaDB cluster

### PR DESCRIPTION
**Pull request by beltran-rubo**
This enables support for configMap for the MariaDB master configuration. I continue working on adding support for the slave.

Important: This also requires a change into the mariadb tar.gz.

### CODE REVIEW

File: incubator/mariadb-cluster/templates/master-deploy.yaml [hunk 1]
```diff
@@ -62,6 +62,7 @@ spec:
         {{- end }}
         - name: MARIADB_DATABASE
           value: "{{ .Values.db.name }}"
+        {{- if .Values.replication.enabled }}
```
Comment (sameersbn) [inline incubator/mariadb-cluster/templates/master-deploy.yaml]: Considering this is a cluster deployment, shouldn't we always enable replication?

File: incubator/mariadb-cluster/values.yaml [hunk 2]
```diff
@@ -61,6 +67,14 @@ master:
     ## Persistent Volume size
     ##
     size: 8Gi
+    ## Configure MySQL with a custom my.cnf file
+    ## ref: https://mysql.com/kb/en/mysql/configuring-mysql-with-mycnf/#example-of-configuration-file
+    ##
+    # config: |-
```
Comment (sameersbn) [inline incubator/mariadb-cluster/values.yaml]: Since we are allowing users  to completely override the `my.ini` configuration, I recommend adding the default configuration here, allowing users to fine tune the configuration.

File: incubator/mariadb-cluster/templates/master-deploy.yaml [hunk 3]
```diff
@@ -86,7 +88,12 @@ spec:
 {{ toYaml .Values.master.resources | indent 10 }}
         volumeMounts:
         - name: data
-          mountPath: /bitnami
+          mountPath: /bitnami/mariadb
+{{- if .Values.master.config }}
```
Comment (sameersbn) [inline incubator/mariadb-cluster/templates/master-deploy.yaml]: IMO the default configuration should be listed in the `values.yaml` file and this condition should be removed

File: incubator/mariadb-cluster/templates/slave-configmap.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{{- if .Values.replication.enabled }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "slave.fullname" . }}
+  labels:
+    app: {{ template "slave.fullname" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
```
Comment (sameersbn) [inline incubator/mariadb-cluster/templates/slave-configmap.yaml]: add `component: "slave"` label

File: incubator/mariadb-cluster/templates/slave-configmap.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
+{{- if .Values.replication.enabled }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "slave.fullname" . }}
+  labels:
+    app: {{ template "slave.fullname" . }}
```
Comment (sameersbn) [inline incubator/mariadb-cluster/templates/slave-configmap.yaml]: `app` label should be `app: "{{ template "name" . }}"`

File: incubator/mariadb-cluster/templates/slave-deploy.yaml [hunk 6]
```diff
@@ -128,8 +134,16 @@ spec:
 {{ toYaml .Values.metrics.resources | indent 10 }}
 {{- end }}
       volumes:
+      - name: config
+        configMap:
+          name: {{ template "slave.fullname" . }}
       - name: data
+      {{- if .Values.slave.persistence.enabled }}
```
Comment (sameersbn) [inline incubator/mariadb-cluster/templates/slave-deploy.yaml]: I wonder whether the persistence setting should instead be cluster wide.

File: incubator/mariadb-cluster/templates/master-deploy.yaml (hunk 1 above)
Comment (beltran-rubo) [reply incubator/mariadb-cluster/templates/master-deploy.yaml]: the idea is to merge this with mariadb-cluster in this repo but later move the changes to the upstream mariadb repo: values.yml would only use a single instance, values-production.yaml would enable replica

File: incubator/mariadb-cluster/templates/slave-deploy.yaml (hunk 6 above)
Comment (beltran-rubo) [reply incubator/mariadb-cluster/templates/slave-deploy.yaml]: well I think for a single node mariadb we should have persistence enabled by default. For the slaves could be optional (though) as it could require a lot of volumes.

File: incubator/mariadb-cluster/values.yaml (hunk 2 above)
Comment (sameersbn) [reply incubator/mariadb-cluster/values.yaml]: the indentation level of the `config` field needs to be fixed  here. Current indentation suggests that its a parameter if of `persistence`

File: incubator/mariadb-cluster/templates/master-deploy.yaml (hunk 1 above)
Comment (sameersbn) [reply incubator/mariadb-cluster/templates/master-deploy.yaml]: Considering that the bitnami/mariadb image only applies the configuration on the first run, I think suggest configuring the master with replication support (replication mode/user/password). So that if a user launches the chart with `replication.enabled=false` and later wants to enable replication and add slaves it should work.

Comment (sameersbn) [reply incubator/mariadb-cluster/templates/master-deploy.yaml]: ... otherwise the users would need to have the foresight and launch the chart with `replication.enabled=true,slave.replicas=0` allowing them to add slaves to the cluster when the need arises.

File: incubator/mariadb-cluster/templates/slave-statefulset.xml [hunk 7]
```diff
@@ -127,25 +140,18 @@ spec:
         resources:
 {{ toYaml .Values.metrics.resources | indent 10 }}
 {{- end }}
-      volumes:
-      - name: data
-        emptyDir: {}
```
Comment (sameersbn) [inline incubator/mariadb-cluster/templates/slave-statefulset.xml]: I suggest using a `enptyDir` volume when persistence is not enabled making it possible to perform some initialization on the data in init-containers if the need ever arises.

File: incubator/mariadb-cluster/values.yaml [hunk 8]
```diff
@@ -61,23 +67,54 @@ master:
...
+    # [mysqld]
+    # skip-name-resolve
+    ##
```
Comment (sameersbn) [inline incubator/mariadb-cluster/values.yaml]: extra comment

File: incubator/mariadb-cluster/values.yaml [hunk 9]
```diff
@@ -61,23 +67,54 @@ master:
...
+    enabled: false
+    # storageClass: "-"
+    annotations:
+    accessModes:
+    - ReadWriteOnce
+    ## Persistent Volume size
+    ##
+    size: 8Gi
+  ## Configure MySQL with a custom my.cnf file
```
Comment (sameersbn) [inline incubator/mariadb-cluster/values.yaml]: adding some new line characters could really  help the readability

File: incubator/mariadb-cluster/templates/master-deploy.yaml (hunk 3 above)
Comment (beltran-rubo) [reply incubator/mariadb-cluster/templates/master-deploy.yaml]: agree

File: incubator/mariadb-cluster/templates/master-configmap.yaml [hunk 10]
```diff
@@ -0,0 +1,14 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "master.fullname" . }}
+  labels:
+    app: {{ template "master.fullname" . }}
```
[the remark this answers is not in the document]
Comment (beltran-rubo) [reply incubator/mariadb-cluster/templates/master-configmap.yaml]: Done

File: incubator/mariadb-cluster/templates/master-deploy.yaml (hunk 1 above)
Comment (beltran-rubo) [reply incubator/mariadb-cluster/templates/master-deploy.yaml]: Agree, done

File: incubator/mariadb-cluster/templates/master-deploy.yaml (hunk 3 above)
Comment (beltran-rubo) [reply incubator/mariadb-cluster/templates/master-deploy.yaml]: Done

File: incubator/mariadb-cluster/templates/slave-configmap.yaml (hunk 5 above)
Comment (beltran-rubo) [reply incubator/mariadb-cluster/templates/slave-configmap.yaml]: done

File: incubator/mariadb-cluster/values-production.yaml [hunk 11]
```diff
@@ -0,0 +1,216 @@
+## Bitnami MariaDB image
+image: beltran/mariadb:latest
```
Comment (juan131) [inline incubator/mariadb-cluster/values-production.yaml]: I guess you set this image for testing purposes. Please remove it from PR

File: incubator/mariadb-cluster/values-production.yaml [hunk 12]
```diff
@@ -0,0 +1,216 @@
...
+pullPolicy: Always
```
Comment (juan131) [inline incubator/mariadb-cluster/values-production.yaml]: Not sure if it's a good idea on production env. variable. I would use `IfNotPresent`

File: incubator/mariadb-cluster/values.yaml [hunk 13]
```diff
@@ -1,6 +1,6 @@
 ## Bitnami MariaDB image
```
Comment (juan131) [inline incubator/mariadb-cluster/values.yaml]: I see a different format on other images:

```
image:
  registry: docker.io
  repository: bitnami/XXXX
  tag: Y.Y.Y
  ## Specify a imagePullPolicy
  ## Defaults to 'Always' if image tag is 'latest', else set to 'IfNotPresent'
  ## ref: http://kubernetes.io/docs/user-guide/images/#pre-pulling-images
  ##
  pullPolicy: Always
  ## Optionally specify an array of imagePullSecrets.
  ## Secrets must be manually created in the namespace.
  ## ref: https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/
  ##
  # pullSecrets:
  #   - myRegistrKeySecretName
```

File: incubator/mariadb-cluster/templates/master-deploy.yaml [hunk 14]
```diff
@@ -4,7 +4,7 @@ metadata:
   name: {{ template "master.fullname" . }}
   labels:
     app: "{{ template "name" . }}"
-    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    chart: {{ template "mariadb.chart" . }}
     component: "master"
     release: "{{ .Release.Name }}"
```
Comment (juan131) [inline incubator/mariadb-cluster/templates/master-deploy.yaml]: You can use instead:

```
{{ .Release.Name | quote }}
```

This apply to every line with format: `""{{ ... }}"`

File: incubator/mariadb-cluster/values-production.yaml (hunk 11 above)
Comment (beltran-rubo) [reply incubator/mariadb-cluster/values-production.yaml]: yes, thanks. I need to integrate the changes in the nami module

File: incubator/mariadb-cluster/values-production.yaml (hunk 12 above)
Comment (beltran-rubo) [reply incubator/mariadb-cluster/values-production.yaml]: agree, thanks

File: incubator/mariadb-cluster/templates/slave-statefulset.xml [hunk 15]
```diff
@@ -1,15 +1,19 @@
-apiVersion: extensions/v1beta1
```
Comment (tompizmor) [inline incubator/mariadb-cluster/templates/slave-statefulset.xml]: rename the file extension to `.yaml`. I don't know why but it `.xml` now

File: incubator/mariadb-cluster/values.yaml (hunk 13 above)
Comment (tompizmor) [reply incubator/mariadb-cluster/values.yaml]: +1, remember to update the master-deploy and slave-statefukset files to retrieve the `image` and `imagePullPolicy` using the new format

File: incubator/mariadb-cluster/templates/master-deploy.yaml [hunk 16]
```diff
@@ -71,22 +72,40 @@ spec:
...
+        {{- if .Values.master.livenessProbe.enabled }}
         livenessProbe:
           exec:
             command: ["sh", "-c", "exec mysqladmin status -uroot -p$MARIADB_ROOT_PASSWORD"]
-          initialDelaySeconds: 120
+          initialDelaySeconds: {{ .Values.master.livenessProbe.initialDelaySeconds }}
+          periodSeconds: {{ .Values.master.livenessProbe.periodSeconds }}
+          timeoutSeconds: {{ .Values.master.livenessProbe.timeoutSeconds }}
+          successThreshold: {{ .Values.master.livenessProbe.successThreshold }}
+          failureThreshold: {{ .Values.master.livenessProbe.failureThreshold}}
```
Comment (tompizmor) [inline incubator/mariadb-cluster/templates/master-deploy.yaml]: Add a space between the last word and the closing braces

File: incubator/mariadb-cluster/templates/slave-configmap.yaml [hunk 17]
```diff
@@ -0,0 +1,15 @@
+{{- if and .Values.replication.enabled .Values.slave.config}}
```
Comment (tompizmor) [inline incubator/mariadb-cluster/templates/slave-configmap.yaml]: add a space before the closing braces

File: incubator/mariadb-cluster/templates/slave-configmap.yaml [hunk 18]
```diff
@@ -0,0 +1,15 @@
...
+  labels:
+    app: {{ template "name" . }}
+    component: "slave"
+    chart: {{ template "mariadb.chart" . }}
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+data:
+  my.cnf: |-
+{{ .Values.slave.config | indent 4 }}
+{{- end -}}
```
Comment (tompizmor) [inline incubator/mariadb-cluster/templates/slave-configmap.yaml]: I think you don't need the second dash: `{{- end }}`

https://github.com/kubernetes/helm/blob/master/docs/chart_template_guide/control_structures.md

> First, the curly brace syntax of template declarations can be modified with special characters to tell the template engine to chomp whitespace. {{- (with the dash and space added) indicates that whitespace should be chomped left, while -}} means whitespace to the right should be consumed. Be careful! Newlines are whitespace!

**Comment by sameersbn** [comment]
@beltran-rubo the PR now has commits to other charts. Can you please fix it

File: incubator/mariadb-cluster/values-production.yaml (hunk 12 above)
Comment (sameersbn) [reply incubator/mariadb-cluster/values-production.yaml]: you'd need to use anyways since we're switching to rolling tags for charts :slightly_smiling_face:

File: incubator/mariadb-cluster/templates/master-deploy.yaml [hunk 19]
```diff
@@ -123,6 +142,9 @@ spec:
 {{ toYaml .Values.metrics.resources | indent 10 }}
 {{- end }}
       volumes:
+      - name: config
```
Comment (sameersbn) [inline incubator/mariadb-cluster/templates/master-deploy.yaml]: Add the following around the `config` volume 

```
{{- if .Values.master.config }}
{{- end }}
```

File: incubator/mariadb-cluster/templates/secrets.yaml [hunk 20]
```diff
@@ -1,28 +1,38 @@
...
   {{- if .Values.db.password }}
   mariadb-password: "{{ .Values.db.password | b64enc }}"
-  {{- else }}
+  {{- else if (not .Values.db.forcePassword) }}
   mariadb-password: "{{ randAlphaNum 10 | b64enc }}"
+  {{- else }}
+  mariadb-password: {{ required "A MariaDB Database Password is required!" .Values.db.password }}
   {{- end }}
   {{- end }}
+  {{- if .Values.replication.enabled }}
```
Comment (sameersbn) [inline incubator/mariadb-cluster/templates/secrets.yaml]: I believe we should remove this `if` statement since we always want to configure the master to support replication

File: incubator/mariadb-cluster/templates/slave-statefulset.yaml [hunk 21]
```diff
@@ -41,16 +45,10 @@ spec:
...
-      {{- range .Values.image.pullSecrets }}
-        - name: {{ . }}
-      {{- end}}
-      {{- end }}
       containers:
       - name: "mariadb"
-        image: "{{ .Values.image.registry }}/{{ .Values.image.repository }}:{{ .Values.image.tag }}"
-        imagePullPolicy: {{ .Values.image.pullPolicy | quote }}
+        image: "{{ .Values.image }}"
+        imagePullPolicy: "{{ .Values.pullPolicy }}"
```
Comment (sameersbn) [inline incubator/mariadb-cluster/templates/slave-statefulset.yaml]: I believe you should revert this change. We're now following the following structure for specifying the chart image

```
image:
  registry: docker.io
  repository: bitnami/mariadb
  tag: 1.2.3
  pullPolicy: Always
```

The elasticsearch chart PR by @juan131 adds a nifty helper for this substitution. It's worth re-using it https://github.com/juan131/charts-1/blob/a9cfe16ed3f5f678fb9b1bae138b5f75739c7f97/incubator/elasticsearch/templates/_helpers.tpl#L25-L32

File: incubator/mariadb-cluster/templates/slave-statefulset.yaml [hunk 22]
```diff
@@ -134,24 +149,22 @@ spec:
...
-    heritage: "{{ .Release.Service }}"
-spec:
-  minReplicas: {{ .Values.slave.hpa.min }}
-  maxReplicas: {{ .Values.slave.hpa.max }}
-  scaleTargetRef:
-    apiVersion: extensions/v1beta1
-    kind: Deployment
-    name: {{ template "slave.fullname" . }}
-  targetCPUUtilizationPercentage: {{ .Values.slave.hpa.target.cpuPercentage }}
+        - name: config
```
Comment (sameersbn) [inline incubator/mariadb-cluster/templates/slave-statefulset.yaml]: add the following around the `config` volume block

```
{{- if .Values.slave.config }}
{{- end }}
```

And restore the `emtyDir` volume for the `data` when persistence is disabled (same as the master)

File: incubator/mariadb-cluster/templates/slave-statefulset.yaml [hunk 23]
```diff
@@ -134,24 +149,22 @@ spec:
...
+          configMap:
+            name: {{ template "slave.fullname" . }}
+
+  volumeClaimTemplates:
+{{- if .Values.slave.persistence.enabled }}
```
Comment (sameersbn) [inline incubator/mariadb-cluster/templates/slave-statefulset.yaml]: While not necessary it makes sense to move `volumeClaimTemplates:` inside the `if` condition

File: incubator/mariadb-cluster/templates/slave-statefulset.yaml [hunk 24]
```diff
@@ -134,24 +149,22 @@ spec:
...
+    - metadata:
+        name: data
```
Comment (sameersbn) [inline incubator/mariadb-cluster/templates/slave-statefulset.yaml]: Please add the standard labels to the template

```
  labels:
    app: "{{ template "name" . }}"
    chart: {{ template "mariadb.chart" . }}
    component: "slave"
    release: "{{ .Release.Name }}"
    heritage: "{{ .Release.Service }}"
```

File: incubator/mariadb-cluster/values-production.yaml (hunk 11 above)
Comment (sameersbn) [reply incubator/mariadb-cluster/values-production.yaml]: also need to adopt the new structure

```
image:
  registry: docker.io
  repository: bitnami/mariadb
  tag: 1.2.3
  pullPolicy: Always
```

File: incubator/mariadb-cluster/values-production.yaml [hunk 25]
```diff
@@ -0,0 +1,216 @@
...
+    periodSeconds: 10
+    timeoutSeconds: 1
+    successThreshold: 1
+    failureThreshold: 3
+
+metrics:
+  enabled: false
+  image: prom/mysqld-exporter
+  imageTag: v0.10.0
+  imagePullPolicy: IfNotPresent
```
Comment (sameersbn) [inline incubator/mariadb-cluster/values-production.yaml]: For consistency, I think we should adopt the new structure here as well

```
metrics:
  image:
    registry: docker.io
    repository: prom/mysqld-exporter
    tag: v0.10.0
    pullPolicy: IfNotPresent
```

File: incubator/mariadb-cluster/values-production.yaml (hunk 12 above)
Comment (juan131) [reply incubator/mariadb-cluster/values-production.yaml]: @sameersbn that's ok for `values.yaml` but we should not use that for `values-production.yaml`

File: incubator/mariadb-cluster/templates/secrets.yaml (hunk 20 above)
Comment (beltran-rubo) [reply incubator/mariadb-cluster/templates/secrets.yaml]: well, it would be enabled by default but I think the user want to have the option to not use replication and a single master, e.g. for testing

File: incubator/mariadb-cluster/values-production.yaml (hunk 12 above)
Comment (sameersbn) [reply incubator/mariadb-cluster/values-production.yaml]: With the new `values-production.yaml`, it gets difficult to know which file you're commenting on.

Yes, you are right. For the prod version we should be using a static tag and pull policy `IfNotPresent`. I withdraw my earlier comment

File: incubator/mariadb-cluster/templates/slave-statefulset.yaml (hunk 21 above)
Comment (tompizmor) [reply incubator/mariadb-cluster/templates/slave-statefulset.yaml]: @beltran-rubo I think you miss this comment

File: incubator/mariadb-cluster/Chart.yaml [hunk 26]
```diff
@@ -1,6 +1,6 @@
 name: mariadb-cluster
-version: 0.5.2
```
Comment (tompizmor) [inline incubator/mariadb-cluster/Chart.yaml]: as you did major changes in this chart you should bump the major version: 
`version: 1.0.0`

File: incubator/mariadb-cluster/templates/slave-svc.yaml [hunk 27]
```diff
@@ -1,12 +1,13 @@
...
 metadata:
   name: {{ template "slave.fullname" . }}
   labels:
     app: "{{ template "name" . }}"
-    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
-    release: "{{ .Release.Name }}"
-    heritage: "{{ .Release.Service }}"
+    chart: {{ template "mariadb.chart" . }}
+    release: {{ .Release.Name | quote }}
+    heritage: {{ .Release.Service | quote }}
```
Comment (sameersbn) [inline incubator/mariadb-cluster/templates/slave-svc.yaml]: `component: slave`

**Comment by sameersbn** [comment]
@beltran-rubo can you fix the user of `{{ .Values.image }}` in the `NOTES.txt`

File: incubator/mariadb-cluster/templates/slave-statefulset.yaml [hunk 28]
```diff
@@ -134,24 +149,38 @@ spec:
...
+            storage: {{ .Values.slave.persistence.size | quote }}
+      {{- if .Values.slave.persistence.storageClass }}
+      {{- if (eq "-" .Values.slave.persistence.storageClass) }}
+        storageClassName: ""
+      {{- else }}
+        storageClassName: {{ .Values.slave.persistence.storageClass | quote }}
+      {{- end }}
+      {{- end }}
+  {{- else }}
+    - name: "data"
```
Comment (sameersbn) [inline incubator/mariadb-cluster/templates/slave-statefulset.yaml]: missing the `volume:` key

Comment (beltran-rubo) [reply incubator/mariadb-cluster/templates/slave-statefulset.yaml]: done

**Comment by beltran-rubo** [comment]
Notes image fixed

**Comment by sameersbn** [comment]
@beltran-rubo i've added a few commits to fix deployments when:
- persistence (slave) is disabled
- metrics exporter is enabled (added helper `metrics.image` for convenience)

**Comment by sameersbn** [comment]
@juan131 @tompizmor i've accepted the PR. You're turn :slightly_smiling_face:

**Comment by sameersbn** [comment]
@beltran-rubo do you think we should bump the number of replicas in the values-production.yaml to a higher number?

**Comment by beltran-rubo** [comment]
Agree, done

File: incubator/mariadb-cluster/values-production.yaml [hunk 29]
```diff
@@ -0,0 +1,233 @@
...
+    - ReadWriteOnce
+    ## Persistent Volume size
+    ##
+    size: 8Gi
+    ##
+
+  ## Configure MySQL slave with a custom my.cnf file
+  ## ref: https://mysql.com/kb/en/mysql/configuring-mysql-with-mycnf/#example-of-configuration-file
+  ##
+  config: |-
```
Comment (javsalgar) [inline incubator/mariadb-cluster/values-production.yaml]: Just a thought out loud for future improvements. Just like we use with secrets, would it make sense to allow the user to use an external config map? 

cc @sameersbn

Comment (sameersbn) [reply incubator/mariadb-cluster/values-production.yaml]: It does make sense, however I've haven't encountered any such requests in the past. If this is something people request then it's worth adding. cc @prydonius

File: incubator/mariadb-cluster/README.md [hunk 30]
```diff
@@ -66,13 +67,41 @@ The following tables lists the configurable parameters of the MariaDB chart and
 | `master.persistence.storageClass` | Persistent Volume Storage Class                     | ``                                                                |
 | `master.persistence.accessModes`  | Persistent Volume Access Modes                      | `[ReadWriteOnce]`                                                 |
 | `master.persistence.size`         | Persistent Volume Size                              | `8Gi`                                                             |
+| `master.config`                   | Config file for the MariaDB Master server           | `_default values in the values.yaml file_`                        |
 | `master.resources`                | CPU/Memory resource requests/limits for master node | `{}`                                                              |
+| `master.livenessProbe.enabled`                    | Turn on and off liveness probe (master)                                                              | `true`                               |               |
```
Comment (juan131) [inline incubator/mariadb-cluster/README.md]: It seems the table is not well aligned. Depends on your editor but the are many modules available (for instance https://atom.io/packages/markdown-table-formatter for ATOM)

File: incubator/mariadb-cluster/templates/NOTES.txt [hunk 31]
```diff
@@ -19,12 +21,15 @@ To connect to your database
...
   2. To connect to master service (read/write):
 
       mysql -h {{ template "master.fullname" . }}.{{ .Release.Namespace }}.svc.cluster.local -uroot -p {{ .Values.db.name }}
 
+{{- if .Values.replication.enabled }}
+
   3. To connect to slave service (read-only):
 
       mysql -h {{ template "slave.fullname" . }}.{{ .Release.Namespace }}.svc.cluster.local -uroot -p {{ .Values.db.name }}
+{{- end }}
```
Comment (juan131) [inline incubator/mariadb-cluster/templates/NOTES.txt]: Currently we added information about how to connect remotely to MariaDB on stable chart and a Waring message. See:

- https://github.com/kubernetes/charts/pull/4869
- https://github.com/kubernetes/charts/blob/master/stable/mariadb/templates/NOTES.txt

Please add it here too.

File: incubator/mariadb-cluster/templates/secrets.yaml [hunk 32]
```diff
@@ -1,28 +1,38 @@
...
   {{- if .Values.replication.password }}
   mariadb-replication-password: "{{ .Values.replication.password | b64enc }}"
-  {{- else }}
+  {{- else if (not .Values.replication.forcePassword) }}
   mariadb-replication-password: "{{ randAlphaNum 10 | b64enc }}"
+  {{- else }}
+  mariadb-password: {{ required "A MariaDB Replication Password is required!" .Values.replication.password }}
```
Comment (tompizmor) [inline incubator/mariadb-cluster/templates/secrets.yaml]: This should be `mariadb-replication-password` right??

**Comment by beltran-rubo** [comment]
Yes thanks, it would be easier for merging changes

File: incubator/mariadb-cluster/templates/secrets.yaml (hunk 32 above)
Comment (sameersbn) [reply incubator/mariadb-cluster/templates/secrets.yaml]: nice catch!
